packages/modules/Extension-Ads/src/worker/RateService.js

```diff
@@ -1,17 +1,22 @@
 export class RateService {
   constructor() {
     this.lastNotificationTimestamp = null;
-    this.initializeLastNotificationTime();
+    this.initialTimestamp = null;
+    this.initializeNotificationTimes();
   }
 
-  async initializeLastNotificationTime() {
+  async initializeNotificationTimes() {
     this.lastNotificationTimestamp = await this.getLastAdsRateNotificationTime();
+    this.initialTimestamp = await this.getInitialNotificationTime();
   }
 
   async shouldNotify() {
     const currentTime = Date.now();
 
-    // 20 min
+    if (!this.lastNotificationTimestamp && (currentTime - this.initialTimestamp) < 1200000) {
```

Comment:
We don't need to store a separate initialTimestamp. We can simply add an if statement and skip the notification if lastNotificationTimestamp is empty.